scene/main/node.cpp

```diff
@@ -2086,9 +2097,10 @@ bool Node::is_ancestor_of(const Node *p_node) const {
 }
 
 bool Node::is_greater_than(const Node *p_node) const {
-	ERR_FAIL_NULL_V(p_node, false);
 	ERR_FAIL_COND_V(!data.tree, false);
 	ERR_FAIL_COND_V(!p_node->data.tree, false);
+	ERR_FAIL_NULL_V(p_node, false);
+	ERR_FAIL_COND_V_MSG(!Thread::is_main_thread(), "is_greater_than can only be called from the main thread.", false);
```

Comment:
This should also be moved up before accessing any data of this node like `data`

Reply:
I’d prefer to avoid changing any methods in this PR and just print an error message , since I’m not sure what regressions it could cause. We can then open issues for 4.6 development where such changes won’t risk regressions.